contracts/lib/RLPReader.sol

```diff
@@ -226,12 +163,18 @@ library RLPReader {
 
     // @return number of payload items inside an encoded list.
     function numItems(RLPItem memory item) private pure returns (uint256) {
-        if (item.len == 0) return 0;
+        // add `isList` check if `item` is expected to be passsed without a check from calling function
+        // require(isList(item), "RLPReader: NUM_ITEMS_NOT_LIST");
```

Comment:
Uncomment?

Reply:
`numItems` is called only from `toList`. And this check is done in that function before calling this one. So this will just add duplicate check in current implementation.